interface.go

```diff
@@ -60,23 +61,30 @@ func (c *Cache) GetOrCompute(key string, evaluator func() (interface{}, error))
 	value, _ := c.computeLocks.LoadOrStore(key, &sync.Mutex{})
 	mutex := value.(*sync.Mutex)
 	mutex.Lock()
-	defer func() {
-		c.computeLocks.Delete(key)
-		mutex.Unlock()
-	}()
 
 	value, err := c.Get(key)
 	if err == nil {
+		c.computeLocks.Delete(key)
+		mutex.Unlock()
 		return value, nil
 	}
 	if err == ErrNotFound {
 		value, err := evaluator()
-		if err != nil {
-			return nil, err
+		if err == nil {
+			go func() {
+				err := c.Set(key, value)
+				c.computeLocks.Delete(key)
+				mutex.Unlock()
+				if err != nil {
+					fmt.Printf("%+v", err)
```

Comment:
Lets put logging of errors to Get() and Set() redis implementations. I don't think there is any other place we can log Set() errors, so lets put it there.